### PR TITLE
IntelliJ/Java: Duplicate code -- Refactored as ExtensionFinder.getClassFromIndex

### PR DESCRIPTION
**Pull request by jsoref**
This is for discussion.

Notes:
* Commits was probably created by IntelliJ for #4034
* In general, commits should not change program behavior at all

### Proposed changelog entries

* `Internal:` Internal Java code cleanup

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers
@Wadeck

### CODE REVIEW

**Review by Wadeck — APPROVED** [review]
Thx for the split, lot easier to review the changes!

nitpick: as you modified the duplicate code, you can ensure consistency by putting `{ ... }` around the last `else` statement (the one for `throw new AssertionError();`) (looking forward to have a better multiline suggestion feature from GH)

**Comment by jsoref** [comment]
@Wadeck: added as a distinct commit because there are two places where that seems useful.

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
It is a protected method in the public class, so formally it is a part of binary API. Please either add Javadoc (with `@since TODO`) or make it private. The rest looks good to me

File: core/src/main/java/hudson/ExtensionFinder.java [hunk 1]
```diff
@@ -758,5 +737,20 @@ private Level logLevel(IndexItem<Extension, Object> item) {
         }
     }
 
+    private static Class<?> getClassFromIndex(IndexItem<Extension, Object> item) throws InstantiationException {
```
Comment (jsoref) [inline core/src/main/java/hudson/ExtensionFinder.java]: @oleg-nenashev: marked as private

**Review by jvz — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
Thanks!

**Pull request merged**